pkg/spdk/server.go

```diff
@@ -206,6 +215,130 @@ func (s *Server) EngineSnapshotDelete(ctx context.Context, req *spdkrpc.Snapshot
 	return &empty.Empty{}, err
 }
 
+func (s *Server) DiskCreate(ctx context.Context, req *spdkrpc.DiskCreateRequest) (*spdkrpc.Disk, error) {
+	log := logrus.WithFields(logrus.Fields{
+		"diskName":  req.DiskName,
+		"diskPath":  req.DiskPath,
+		"blockSize": req.BlockSize,
+	})
+
+	log.Info("Creating disk")
+
+	if req.DiskName == "" || req.DiskPath == "" {
+		return nil, grpcstatus.Error(grpccodes.InvalidArgument, "disk name and disk path are required")
+	}
+
+	s.Lock()
+	defer s.Unlock()
+
+	if err := s.validateDiskCreateRequest(req); err != nil {
+		log.WithError(err).Error("Failed to validate disk create request")
+		return nil, grpcstatus.Error(grpccodes.InvalidArgument, errors.Wrap(err, "failed to validate disk create request").Error())
+	}
+
+	blockSize := uint64(defaultBlockSize)
+	if req.BlockSize > 0 {
+		log.Infof("Using custom block size %v", req.BlockSize)
+		blockSize = uint64(req.BlockSize)
+	}
+
+	uuid, err := s.addBlockDevice(req.DiskName, req.DiskPath, blockSize)
+	if err != nil {
+		log.WithError(err).Error("Failed to add block device")
+		return nil, grpcstatus.Error(grpccodes.Internal, errors.Wrap(err, "failed to add block device").Error())
+	}
+
+	return s.lvstoreToDisk(req.DiskPath, "", uuid)
+}
+
+func (s *Server) DiskDelete(ctx context.Context, req *spdkrpc.DiskDeleteRequest) (*emptypb.Empty, error) {
+	log := logrus.WithFields(logrus.Fields{
+		"diskName": req.DiskName,
+		"diskUUID": req.DiskUuid,
+	})
+
+	log.Info("Deleting disk")
+
+	if req.DiskName == "" || req.DiskUuid == "" {
+		return &empty.Empty{}, grpcstatus.Error(grpccodes.InvalidArgument, "disk name and disk UUID are required")
+	}
+
+	s.Lock()
+	defer s.Unlock()
+
+	lvstores, err := s.spdkClient.BdevLvolGetLvstore("", req.DiskUuid)
+	if err != nil {
+		resp, parseErr := parseErrorMessage(err.Error())
+		if parseErr != nil || !isNoSuchDevice(resp.Message) {
+			return nil, errors.Wrapf(err, "failed to get lvstore with UUID %v", req.DiskUuid)
+		}
+		log.WithError(err).Errorf("Cannot find lvstore with UUID %v", req.DiskUuid)
+		return nil, grpcstatus.Errorf(grpccodes.NotFound, "cannot find lvstore with UUID %v", req.DiskUuid)
```

Comment:
Can we continue deleting the aio bdev when lvstore is not found?

Reply:
Sure.